src/views/document/RouteView.vue

```diff
@@ -180,6 +185,27 @@ export default {
 
       return result;
     },
+    showMissingHistoryBanner() {
+      const doc = this.document;
+      const activities = doc.activities ?? [];
+      const history_worth_activities = [
```

Comment:
to avoid computing the array each time, you could move the declaration out of the default export (agreed, very small perf improvement)